src/setup_payload/AdditionalDataPayloadParser.cpp

```diff
@@ -42,20 +43,27 @@ CHIP_ERROR AdditionalDataPayloadParser::populatePayload(SetupPayloadData::Additi
     TLV::ContiguousBufferTLVReader innerReader;
 
     reader.Init(mPayloadBufferData, mPayloadBufferLength);
-    ReturnErrorOnFailure(reader.Next(TLV::kTLVType_Structure, TLV::AnonymousTag));
+    ReturnErrorOnFailure(reader.Next(TLV::kTLVType_Structure, RotatingDeviceId::DiscoveryExtensionDescriptorTag));
 
     // Open the container
     ReturnErrorOnFailure(reader.OpenContainer(innerReader));
+    if (innerReader.Next(TLV::kTLVType_ByteString, TLV::ContextTag(SetupPayloadData::kRotatingDeviceIdTag)) == CHIP_NO_ERROR)
+    {
+        // Get the value of the rotating device id
+        ByteSpan rotatingDeviceId(innerReader.GetReadPoint(), innerReader.GetLength());
```

Comment:
This line of code is a critical security bug.

Please use `GetByteView`.

And please add tests that would have caught the bug.

Reply:
good catch, i changed it to GetByteView and added a unit test with a longer rotating id length than expected to make sure we don't read beyond TLV memory limits